crates/core/src/models/transaction/transaction.rs

```diff
@@ -103,7 +103,7 @@ impl ConvertibleStarknetTransaction for StarknetTransaction {
             },
             _ => return Err(EthApiError::KakarotDataFilteringError("Transaction".into())),
         };
-        let nonce: U256 = nonce.into();
+        let nonce: U64 = u64::try_from(nonce).expect("INVARIANT: nonce should be a u64").into();
```

Comment:
This looks very weird, why are we doing this? I don't think the RPC should be doing the job of checking if nonces are u64

Reply:
Eth Api expects a U64, whereas Starknet RPC returns a FieldElement, so we need to make one fit into the other

Reply:
Ok will change for "?"